ambari-server/src/main/java/org/apache/ambari/server/stack/StackManager.java

```diff
@@ -188,6 +188,29 @@ public StackManager(@Assisted("stackRoot") File stackRoot,
     populateDB(stackDao, extensionDao);
   }
 
+  /***
+   *  Constructor. Initialize StackManager for merging service definitions and creating merged stack definition
+   * @param stackRoot
+   * @param commonServicesRoot
+   */
+  public StackManager(File stackRoot, File commonServicesRoot, boolean validate) throws AmbariException {
```

Comment:
I feel it's dangerous to use a class which is supposed to be `@Inject` without injection. It's very possible that the addition of fields in the Injected constructor could cause NPEs when the utility goes to use this constructor.